### PR TITLE
OCPBUGS-3276: Pin Down Dependencies on release 4.12

### PR DESCRIPTION
**Pull request by raptorsun**
* [ ] I added CHANGELOG entry for this change.
* [x] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,43 +6,43 @@ require (
...
+	k8s.io/utils v0.0.0-20221012122500-cfd413dd9e85
 )
 
 require (
-	github.com/prometheus/common v0.34.0
+	github.com/prometheus/common v0.37.0
 	gopkg.in/yaml.v3 v3.0.1
 )
 
+replace github.com/prometheus/prometheus => github.com/prometheus/prometheus v0.39.1
```
Comment (jan--f) [inline go.mod:38]: Why the `replace` statement? Do we want a reminder here about its purpose?

Comment (raptorsun) [reply go.mod:38]: yes, it is to replace the version 1.99 of  github.com/prometheus/prometheus in the require section above.

Comment (jan--f) [reply go.mod:38]: Right, but can you leave a little note here to explain why we need that please?

Comment (raptorsun) [reply go.mod:38]: of course 👍

Comment (raptorsun) [reply go.mod]: updated now

**Comment by raptorsun** [comment]
Prometheus-operator has changed the API structure of `PrometheusWebSpec`. An adaptation in the e2e test code is made.

**Comment by raptorsun** [comment]
/hold
we are doing this using backport process.
see https://github.com/openshift/cluster-monitoring-operator/pull/1814
and https://issues.redhat.com/browse/OCPBUGS-3331

**Comment by raptorsun** [comment]
/close
This PR is closed in favor of the merged PR https://github.com/openshift/cluster-monitoring-operator/pull/1814

**Pull request closed without merging**